src/main/java/com/jadc/bazaar/controller/NotificationController.java

```diff
@@ -139,39 +140,30 @@ public String save(
 			@ModelAttribute("notification") Notification notification,
 			@RequestParam("publicationDateTime") String publicationDateTime,
 			@RequestParam("notificationCategory") int category,
-			Model model) {
+			RedirectAttributes attributes) {
 
 		NotificationCategory notificationCategory = notificationCategoryRepository.findById(category).orElseThrow();
 
 		notification.setNotificationCategory(notificationCategory);
 		notification.setIsDeleted(false);
 
-		String action;
-		if (notification.getId() > 0) {
-			action = "updated";
-		} else {
-			action = "created";
-		}
-
 		try {
 			notification.setPublicationDate(LocalDateTime.parse(publicationDateTime));
-
-			model.addAttribute("alertMsg", notification.getTitle()+" has been "+action);
+			attributes.addFlashAttribute("saveSuccess", true);
 
 			notificationService.save(notification);
-
-			return root(model);
 		} catch (DateTimeParseException e) {
-			model.addAttribute("errorMsg", "An error occurred in "+action+" notification.");
-
-			return root(model);
+			attributes.addFlashAttribute("saveError", true);
 		}
+
+		return "redirect:/{lang}/admin/notification";
 	}
 
 	@PostMapping("/delete")
-	public String deleteRows(@RequestParam("notifications") Integer[] notifications) {
+	public String deleteRows(@RequestParam("notifications") Integer[] notifications, RedirectAttributes attributes) {
 		notificationService.delete(notifications);
+		attributes.addFlashAttribute("deleteSuccess", true);
 
-		return "redirect:";
+		return "redirect:/{lang}/admin/notification";
```

Comment:
The {lang}/ comes from the URL path, Sir. "en/" or "ja/"
`http://localhost:8080/en/admin/account`
`http://localhost:8080/ja/admin/account`

They are integrated with the href of the links in the template, which is retrieved from the properties (app.message.locale):
`<a th:href="@{'/'+#{app.message.locale}+'/admin/account'}"`

I think /{lang}/ will always be set. If the URL has not specified/populated with "en" or "ja", it will redirect to /ja/** since the default locale is ja.